apps/web/src/pages/Pools/PoolAdvertisementPage/components/DeadlineDialog.tsx

```diff
@@ -0,0 +1,159 @@
+import React from "react";
+import { useIntl } from "react-intl";
+import InformationCircleIcon from "@heroicons/react/24/solid/InformationCircleIcon";
+
+import { Button, Dialog, Link } from "@gc-digital-talent/ui";
+import { formatDate } from "@gc-digital-talent/date-helpers";
+
+import useRoutes from "~/hooks/useRoutes";
+
+const generateLink = (href: string, chunks: React.ReactNode) => (
+  <Link newTab external href={href}>
+    {chunks}
+  </Link>
+);
+
+type DeadlineDialogProps = {
+  deadline: Date;
+};
+
+const DeadlineDialog = ({ deadline }: DeadlineDialogProps): JSX.Element => {
+  const intl = useIntl();
+  const routes = useRoutes();
+  // https://stackoverflow.com/a/34602679
+  const localTimeZone = Intl.DateTimeFormat().resolvedOptions().timeZone;
+
+  return (
+    <Dialog.Root>
+      <Dialog.Trigger>
+        <Button
+          mode="inline"
+          color="secondary"
+          icon={InformationCircleIcon}
+          aria-label={intl.formatMessage({
+            defaultMessage: "Learn about how application deadlines work.",
+            id: "8YKsal",
+            description:
+              "Info button label for pool application deadline details.",
+          })}
+        />
```

Comment:
We could make this a mode variant?

Reply:
I gave it a swing.
[Add icon_only button variant](https://github.com/GCTC-NTGC/gc-digital-talent/pull/8544/commits/ada899598521f052d3b0b36920b8cc03fb96a1d5)

Reply:
I'll let @substrae make the final decision but this looks great to me! 👏

Reply:
@petertgiles Sorry I forgot about this! This looks good at first glance - however I did notice the dark mode hover effects don't meet our contrast requirements. Chances are we'll need to flip the colors on hover so that the ring continues to be the lighter value and the icon turns to darker. This is also relevant to the `white` color combo, because the hover is using `secondary` values on a black background. Working on the buttons is a bit of a beast, because there are so many color combinations to worry about. Let me know if you need a hand and we can pair on it.

Reply:
Before I forget - any contrast fixes should match the existing palettes and patterns used by the buttons already. 🙈

Reply:
OK, I think I've gotten it better.   IAP light mode black and white have a very subtle effect - maybe that one needs to be tweaked because it can't use the secondary colour?  Feel free to make any changes you'd like.  :grin: 
[better contrast for buttons](https://github.com/GCTC-NTGC/gc-digital-talent/pull/8544/commits/0bfff0517c52e7325b2e4f3aec2bdfc5aa298cc2)

Reply:
[tweak lighter colors](https://github.com/GCTC-NTGC/gc-digital-talent/pull/8544/commits/e08cb854b9ff8d50d37348d669565361d4560b63)

Reply:
Awesome, thanks for fixing it up @petertgiles